### DISCUSSION TITLE
environment specific headers

### DISCUSSION THREAD

**Discussion by splusq** [AnWeber/httpyac#611] [General]
I am trying to workout a scheme where `defaultHeaders` are set based on environment. I tried something like this, but it didn't quiet work. Any ideas on what I could do? 

```
module.exports = {
  "environments": {
    "$default": {
      "host": "https://foo",
      "headers": {
        "bar": "baz"
      }
    },
    "koo": {
      "host": "https://koo",
      "headers": {
        "bar": "baz2",
        "bar2": "bar2",
      }
    }
  },
  "defaultHeaders": headers
}
```

**Comment by AnWeber** [comment]
I don't quite have the use case. However, I have a special plugin on my work computer that sets certain headers depending on the domain. With `HTTPYAC_PLUGIN` you can configure a global script for the current system that is always executed on every request. The file does not have to be global and is also automatically found in the current project. In this script you could access your current env variables and add it to the request headers. 

e.g HTTPYAC_PLUGIN=~/.httpyac.js

```javascript
module.exports = {
  configureHooks: function (api) {
    api.hooks.onRequest.addHook('addHeader', function (request, {variables}) {
      if(request.url.indexOf("example.com") >= 0){
        request.headers.authorization = `Bearer ${variables.token}`;
      }
    });
  }
}
```

Any good ideas on how I can generalize this? I just haven't had a good way how to define env specific default headers.
